src/tokenizer_test.cc

```diff
@@ -77,6 +78,18 @@ TEST_F(TokenizerTest, ProcessDouble) {
   EXPECT_TRUE(next->IsEOS());
 }
 
+TEST_F(TokenizerTest, ProcessNaN) {
+  Tokenizer t("NaN");
```

Comment:
Can you make this a parameterized test and try different combinations? NaN, nan, NAN, etc?

Reply:
Will do.